policy-controller/core/src/outbound.rs

```diff
@@ -42,6 +42,21 @@ pub type GrpcRoute = OutboundRoute<GrpcRouteMatch, GrpcRetryCondition>;
 
 pub type RouteSet<T> = HashMap<GroupKindNamespaceName, T>;
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum AppProtocol {
+    Opaque,
+    Unknown(Arc<str>),
+}
+
+impl From<&str> for AppProtocol {
+    fn from(value: &str) -> Self {
+        match value.to_ascii_lowercase().as_str() {
+            "linkerd.io/tcp" | "linkerd.io/opaque" => AppProtocol::Opaque,
+            protocol => AppProtocol::Unknown(Arc::from(protocol.to_string().into_boxed_str())),
```

Comment:
```suggestion
            protocol => AppProtocol::Unknown(Arc::from(protocol)),
```

I don't think these two allocations are necessary, are they?

Reply:
I believe it keeps the same allocation through the conversions, but your suggestion is cleaner and clearer anyway.